src/vs/workbench/contrib/testing/browser/testingExplorerView.ts

```diff
@@ -81,8 +82,10 @@ export class TestingExplorerView extends ViewPane {
 	private filterActionBar = this._register(new MutableDisposable());
 	private container!: HTMLElement;
 	private treeHeader!: HTMLElement;
+	private countSummary!: CountSummary;
```

Comment:
This isn't defined at a known point in the lifecycle. I would remove this property and just pass the summary as an argument to `renderActivityCount`

Reply:
Passing the summary as an argument won't make the badge reactive when there's a change in the configurations, that's why I thought about keeping the summary in the property, so whenever the configuration is updated the render method can find the summary. 
```
this._register(onDidChangeTestingCountBadge(this.renderActivityCount, this));
```

But thinking about it, do we really want to make it that reactive? I mean, if we just need to react only if it went from any type to `off`, we can pass undefined maybe as an argument to the function and it would work. but if we want to make it totally reactive, like if you run the tests with badge type `passed` and then changed it to `failed`, what do you expect? do you expect to see the number updated immediately?

what do you think?

Reply:
Ah, I see, missed that usage. I would then just remove the `!` and type it as `CountSummary | undefined`

Reply:
done, added a guard check also in the render method, are we fine with that?